### DISCUSSION TITLE
책 검색 시 API 호출 주기 질문

### DISCUSSION THREAD

**Discussion by tinajeong** [prgrms-web-devcourse/Team-Gaerval-Dadok-FE#2] [General]
안녕하세요!

프론트팀에서 책 검색 시 API 호출 주기를 어떻게 처리하실 계획인지 여쭤보고 싶습니다.

백앤드팀이 살펴본 결과, 서비스마다 검색 시 API를 호출하는 주기가 상이한 듯합니다. 
타이핑시마다 호출하는 서비스도 있고, 검색어를 입력한 후 클릭이벤트가 발생해야 호출하는 케이스도 있네요.

어떤 방식으로 하시든지 무관하지만 어느정도 호출횟수를 줄여 서버부하를 줄이는 방향이면 좋을 것 같습니다! 😉
ex) 타이핑 시 마다 호출하지만 일정 시간 이후에 호출

**Comment by harryk-im** [comment]
프론트쪽에서는 타이핑 이벤트의 디바운싱 시간을 1초로 생각 하고 있습니다!
